instrumentation/hibernate/hibernate-6.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/hibernate/v6_0/EntityManagerTest.java

```diff
@@ -0,0 +1,331 @@
+                "createSQLQuery",
+                new Parameter("createSQLQuery", "SELECT Value", queryBuildMethods.get(2)))));
+  }
+
+  private static class Parameter {
+
+    Parameter(String methodName, String resource, Function<EntityManager, Query> queryBuildMethod) {
+      this.methodName = methodName;
+      this.resource = resource;
+      this.attach = false;
+      this.flushOnCommit = false;
+      this.sessionMethodTest = null;
+      this.queryBuildMethod = queryBuildMethod;
+    }
+
+    Parameter(
+        String methodName,
+        String resource,
+        boolean attach,
+        boolean flushOnCommit,
+        BiConsumer<EntityManager, Value> sessionMethodTest) {
+      this.methodName = methodName;
+      this.resource = resource;
+      this.attach = attach;
+      this.flushOnCommit = flushOnCommit;
+      this.sessionMethodTest = sessionMethodTest;
+      this.queryBuildMethod = null;
+    }
+
+    public final String methodName;
+    public final String resource;
+    public final boolean attach;
+    public final boolean flushOnCommit;
+    public final BiConsumer<EntityManager, Value> sessionMethodTest;
+    public final Function<EntityManager, Query> queryBuildMethod;
+  }
+
+  private static SpanDataAssert assertClientSpan(SpanDataAssert span, SpanData parent) {
+    return span.hasKind(SpanKind.CLIENT)
+        .hasParent(parent)
+        .hasAttributesSatisfyingExactly(
+            equalTo(SemanticAttributes.DB_SYSTEM, "h2"),
+            equalTo(SemanticAttributes.DB_NAME, "db1"),
+            equalTo(SemanticAttributes.DB_USER, "sa"),
+            equalTo(SemanticAttributes.DB_CONNECTION_STRING, "h2:mem:"),
+            satisfies(SemanticAttributes.DB_STATEMENT, val -> val.isInstanceOf(String.class)),
+            satisfies(SemanticAttributes.DB_OPERATION, val -> val.isInstanceOf(String.class)),
+            equalTo(SemanticAttributes.DB_SQL_TABLE, "Value"));
+  }
+
+  private static SpanDataAssert assertClientSpan(
+      SpanDataAssert span, SpanData parent, String spanName) {
+    return span.hasName(spanName)
+        .hasKind(SpanKind.CLIENT)
+        .hasParent(parent)
+        .hasAttributesSatisfyingExactly(
+            equalTo(SemanticAttributes.DB_SYSTEM, "h2"),
+            equalTo(SemanticAttributes.DB_NAME, "db1"),
+            equalTo(SemanticAttributes.DB_USER, "sa"),
+            equalTo(SemanticAttributes.DB_CONNECTION_STRING, "h2:mem:"),
+            satisfies(SemanticAttributes.DB_STATEMENT, val -> val.isInstanceOf(String.class)),
+            satisfies(SemanticAttributes.DB_OPERATION, val -> val.isInstanceOf(String.class)),
+            equalTo(SemanticAttributes.DB_SQL_TABLE, "Value"));
+  }
+
+  private static SpanDataAssert assertSessionSpan(
+      SpanDataAssert span, SpanData parent, AtomicReference<String> sessionId, String spanName) {
+    return span.hasName(spanName)
+        .hasKind(SpanKind.INTERNAL)
+        .hasParent(parent)
+        .hasAttributesSatisfying(
```

Comment:
could this be replaced with `hasAttributesSatisfyingExactly`? Perhaps it would be better to capture the session id in the caller, I think you could uses something like `trace.getSpan(1).getAttributes().get(AttributeKey.stringKey("hibernate.session_id"))`

Reply:
Done. I have replaced with `hasAttributesSatisfyingExactly` for all tests. removed sessionid and used `trace.getSpan(1).getAttributes().get(AttributeKey.stringKey("hibernate.session_id"))`  in all tests . Please review.